### PR TITLE
Added: support to set current location from settings page (#238pmgz).

### PR DESCRIPTION
**Pull request by meet-aniket**

### CODE REVIEW

File: src/store/modules/user/actions.ts [hunk 1]
```diff
@@ -87,6 +107,26 @@ const actions: ActionTree<UserState, RootState> = {
...
+      if(resp.status === 200 && resp.data.count > 0 && !hasError(resp)) {
+        let facilityLocations = resp.data.docs
+        facilityLocations = facilityLocations.map((location: any) => {
+          return {
+            locationSeqId: location.locationSeqId,
+            locationPath: location.areaId + location.aisleId + location.sectionId + location.levelId + location.positionId
+          }
+        })
+
+        return facilityLocations;
```
Comment (ymaheshwari1) [inline src/store/modules/user/actions.ts:125]: We should also return a value when we have an error in response.

File: src/store/modules/user/actions.ts [hunk 2]
```diff
@@ -52,14 +52,30 @@ const actions: ActionTree<UserState, RootState> = {
...
       const localTimeZone = moment.tz.guess();
       if (resp.data.userTimeZone !== localTimeZone) {
         emitter.emit('timeZoneDifferent', { profileTimeZone: resp.data.userTimeZone, localTimeZone});
       }
+      if(resp.data.facilities.length > 0) {
+        const payload = {
+          "inputFields": {
+            "facilityId": resp.data.facilities[0].facilityId
+          },
+          "fieldList": [],
```
Comment (ymaheshwari1) [inline src/store/modules/user/actions.ts:67]: I think we should add fields if we just want to use some specific fields.

File: src/store/modules/user/actions.ts [hunk 3]
```diff
@@ -70,6 +86,10 @@ const actions: ActionTree<UserState, RootState> = {
   async setFacility ({ commit }, payload) {
     commit(types.USER_CURRENT_FACILITY_UPDATED, payload.facility);
```
Comment (ymaheshwari1) [inline src/store/modules/user/actions.ts:87]: We also need to fetch the facilityLocation whenever the current facility is changed.

File: src/views/Settings.vue [hunk 4]
```diff
@@ -17,6 +17,14 @@
          </ion-select>
        </ion-item>
 
+       <ion-item>
+         <ion-icon :icon="locationOutline" slot="start" />
+         <ion-label>{{$t("Location")}}</ion-label>
+         <ion-select interface="popover" :placeholder="$t('facility location')" :value="currentFacilityLocation.locationSeqId" @ionChange="setFacilityLocation($event)">
+           <ion-select-option v-for="facilityLocation in (userProfile && userProfile.facilityLocations ? userProfile.facilityLocations : [])" :key="facilityLocation.locationSeqId" :value="facilityLocation.locationSeqId" >{{ facilityLocation.locationPath }}</ion-select-option>
```
Comment (ymaheshwari1) [inline src/views/Settings.vue:24]: I don't think `facilityLocation` has property `locationPath` with it, let me know if I missed something.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/user/actions.ts (hunk 1 above)
Comment (meet-aniket) [reply src/store/modules/user/actions.ts:125]: Returned an empty array when getting any error or not getting proper response from api.

File: src/store/modules/user/actions.ts (hunk 2 above)
Comment (meet-aniket) [reply src/store/modules/user/actions.ts:67]: Added all fields to fetch from API in fieldList.

File: src/store/modules/user/actions.ts (hunk 3 above)
Comment (meet-aniket) [reply src/store/modules/user/actions.ts:87]: Updated code to fetch the facilityLocation whenever the current facility is changed.

File: src/views/Settings.vue (hunk 4 above)
Comment (meet-aniket) [reply src/views/Settings.vue:24]: Yes Sir, we are not getting this field with API instead we are creating this field after getting response back from API.
https://github.com/hotwax/receiving/pull/89/files#:~:text=locationSeqId%3A%20location.locationSeqId,levelId%20%2B%20location.positionId

**Comment by azkyakhan** [comment]
Note:

Will use spacer variables in another PR.

**Comment by azkyakhan** [comment]
Note:

Will use spacer variables in another PR.

**Pull request merged**